geopandas/geodataframe.py

```diff
@@ -1,18 +1,37 @@
+from collections import Iterable
 import json
 import os
 import sys
 
 import numpy as np
+import pandas as pd
+from pandas.core.internals import BlockManager
 from pandas import DataFrame, Series, Index
 from shapely.geometry import mapping, shape
 from shapely.geometry.base import BaseGeometry
 from six import string_types, PY3
 
-from geopandas import GeoSeries
+from .geoseries import GeoSeries
 from geopandas.base import GeoPandasBase
 from geopandas.plotting import plot_dataframe
 import geopandas.io
 
+from . import vectorized
+from ._block import GeometryBlock
+
+
+def coerce_to_geoseries(x, **kwargs):
+    if isinstance(x, GeoSeries):
+        return x
+    if isinstance(x, vectorized.VectorizedGeometry):
+        return GeoSeries(x, **kwargs)
+    if isinstance(x, pd.Series):
+        return GeoSeries(vectorized.from_shapely(list(x)),
```

Comment:
is the conversion to list needed? I suppose `from_shapely` also works on an array?